### PR TITLE
WT-3266 Drop lock while joining threads.

### PR DESCRIPTION
**Pull request by sueloverso**
@keithbostic or @agorrod Do you want to review this?

### CODE REVIEW

**Comment by sueloverso** [comment]
This fix has run more than 3x the iterations successfully (over 2000 so far) than the original repro (602).

File: src/support/thread_group.c [hunk 1]
```diff
@@ -23,6 +23,7 @@ __thread_run(void *arg)
 	session = thread->session;
 
 	for (;;) {
+		WT_ERR(WT_SESSION_CHECK_PANIC(session));
```
Comment (keithbostic) [inline src/support/thread_group.c]: I don't understand why we're checking panic here -- at best I don't think it makes anything better. There's some related discussion in WT-3245.

File: src/support/thread_group.c [hunk 2]
```diff
@@ -92,8 +93,27 @@ __thread_group_shrink(
 		 */
```
Comment (keithbostic) [inline src/support/thread_group.c]: This comment looks strange to me, preceding a bunch of code that has nothing to do with waking threads? (I realize it's in the original code, but figured I'd mention it.)

File: src/support/thread_group.c (hunk 1 above)
Comment (sueloverso) [inline src/support/thread_group.c]: If the system panics, I don't see anything that will break this loop.  However, I'm happy to let that be considered under WT-3245 with other threads too.

**Comment by sueloverso** [comment]
I completed 10000 iterations of this branch with the CONFIG.

**Comment by michaelcahill** [comment]
Thanks @sueloverso, this change looks right to me.  I'll merge.

**Pull request merged**